### ISSUE TITLE
Avoid wiping quickfix list on `:RescriptFormat`

### ISSUE THREAD

**Issue by sinap-fb** [rescript-lang/vim-rescript#52]
(This probably applies to the `:RescriptBuild` command as well)

On format, the users quickfix list should not be wiped, since it can be a list of stuff unrelated to Rescript.

Currently, it is getting wiped here:
https://github.com/rescript-lang/vim-rescript/blob/08de54132587131e762b036c11e0e9a9603992fa/autoload/rescript.vim#L164

 I think this should be achievable using quickfix IDs.

Thanks

**Comment by ryyppy** [comment]
yeah, i made this because I didn't know how to isolate quickfix results to certain scope. Didn't know about ids... currently I don't consider this a very big problem (except for wiping e.g. build errors from quickfix.. but for that I usually just rerun the build real quick with a hotkey). 

Don't have much time right now to fix this right away, so I would be open for a PR!
